src/util/expr.h

```diff
@@ -84,6 +84,19 @@ class exprt:public irept
     return static_cast<const typet &>(find(ID_type));
   }
 
+  /// Amount of nodes this expression tree contains.
+  std::size_t size() const
```

Comment:
Document that complexity is linear in the tree size, so that we are warned against using it frivolously.